clang-tools-extra/clangd/unittests/IncludeCleanerTests.cpp

```diff
@@ -135,6 +135,21 @@ TEST(IncludeCleaner, GetUnusedHeaders) {
                            Pointee(writtenInclusion("\"dir/unused.h\""))));
 }
 
+TEST(IncludeCleaner, SystemUnusedHeaders) {
+  auto TU = TestTU::withCode(R"cpp(
+    #include <system_header.h>
+    #include <system_unused.h>
+    SystemClass x;
+  )cpp");
+  TU.AdditionalFiles["system/system_header.h"] = guard("class SystemClass {};");
+  TU.AdditionalFiles["system/system_unused.h"] = guard("");
+  TU.ExtraArgs = {"-isystem", testPath("system")};
+  auto AST = TU.build();
+  IncludeCleanerFindings Findings = computeIncludeCleanerFindings(AST, true);
```

Comment:
can you also add a negative test? (i.e. when `AnalyzeSystemHeaders = false`, `UnusedIncludes` is empty)

Reply:
Isn;t this already covered by the `GetUnusedHeaders` test? There's an unused `<system_header.h>` which must not present in `UnusedIncludes`.

Reply:
> Isn;t this already covered by the GetUnusedHeaders test?

Yes but that one is a more bulky test that checks for multiple pieces playing nicely together. Whereas this one is more "specific" to the code-path you're adding.